src/modules/0.0.14/aws_ecs_fargate/sql/create_custom_constraints.sql

```diff
@@ -12,3 +12,29 @@ begin
 end;
 $$;
 ALTER TABLE service ADD CONSTRAINT check_service_subnets CHECK (check_service_subnets(subnets));
+
+CREATE OR REPLACE FUNCTION check_subnets_by_service() RETURNS trigger AS $check_subnets_by_service$
+    DECLARE
+        service_row record;
+    BEGIN
+        FOR service_row IN
+            SELECT * FROM service
+        LOOP
+            IF OLD.subnet_id = any(service_row.subnets) THEN
+                RAISE EXCEPTION 'subnet_id % is being used by service %', OLD.subnet_id, service_row.name;
+            END IF;
+        END LOOP;
+        IF (TG_OP = 'DELETE') THEN
```

Comment:
Am I reading this correctly that deletes are basically banned at the database level here?

Reply:
not sure what do you mean?  I'm executing this function on a `BEFORE DELETE OR UPDATE` trigger, the only difference is that on `DELETE` the trigger needs to return the `OLD` record and for `UPDATE`s need to return the `NEW` record

Reply:
Oh no no, it is blocking `UPDATE`s and `DELETE`s if the subnet id in question is present in any of the service `subnets` column